### PR TITLE
SLES4SAP Tests Improvements

### PR DESCRIPTION
**Pull request by alvarocarvajald**
This PR includes the following:

1. Replaces the call to `wait_still_screen` in `tests/sles4sap/netweaver_ascs_install` with a call to ~check_screen 'root-console'~ `assert_screen` with multiple tags. The call to `systemctl restart systemd-logind` does reset the screen away from `root-console` when `DESKTOP=gnome`, but instead of waiting for the screen to stop from changing, a call to verify if the `root-console` is correctly selected followed by a `select_console 'root-console'` also fixes the test, while improving readibility. Also, when the test is run in a system with `DESKTOP=textmode`, ~check_screen~ `assert_screen` should return faster as the `root-console` is not changed instead of the 7 seconds of still screen in `wait_still_screen`, while when `DESKTOP=gnome` the default timeout of ~check_screen~ `assert_screen` is the same as the timeout of `wait_still_screen`.
2. Improve regexp in `tests/sles4sap/sapconf` to avoid false positives when the serial console is contaminated with extra characters.
3. Disable `packagekit` before `zypper` calls in `tests/sles4sap/patterns`. ~Re-enable after test is done~. This in order to fix these false positives: https://openqa.suse.de/tests/1485695#step/patterns/27

- Related ticket: N/A
- Needles: N/A
- Verification run: http://mango.suse.de/tests/245 & http://mango.suse.de/tests/246

### CODE REVIEW

File: tests/sles4sap/patterns.pm [hunk 1]
```diff
@@ -22,6 +23,9 @@ sub run {
 
     select_console 'root-console';
 
+    # Disable packagekit
+    systemctl '--now mask packagekit.service';
```
Comment (asmorodskyi) [inline tests/sles4sap/patterns.pm]: better to use pkcon_quit search for it in utils

File: tests/sles4sap/patterns.pm [hunk 2]
```diff
@@ -51,6 +55,9 @@ sub run {
         die "SAP zypper pattern [$pattern] info check failed"
           unless ($output =~ /i\+\s\|\spatterns-$pattern\s+\|\spackage\s\|\sRequired/);
     }
+
+    # Enable packagekit
+    systemctl '--now unmask packagekit.service';
```
Comment (asmorodskyi) [inline tests/sles4sap/patterns.pm]: any good reason to enable it ?

File: tests/sles4sap/netweaver_ascs_install.pm [hunk 3]
```diff
@@ -52,8 +52,7 @@ sub run {
 
     # If running in DESKTOP=gnome, systemd-logind restart may cause the graphical console to
     # reset and appear in SUD, so need to select 'root-console' again
-    wait_still_screen;
-    select_console 'root-console';
+    select_console 'root-console' unless (check_screen 'root-console');
```
Comment (okurz) [inline tests/sles4sap/netweaver_ascs_install.pm]: Are you sure this saves time? By default the timeout of `check_screen` is 30 seconds and not really the same behaviour as wait_still_screen. It is more likely for a screen to show up as still than exactly matching one tag so I fear you are wasting 30s here. Also, I do not recommend to use `check_screen` with default timeout anywhere. A better alternative is either `check_screen …, 0` which I think won't work here as you tell that systemd-logind switches away the console *or* a multi-tag assert_screen with `match_has_tag` which might work here

File: tests/sles4sap/patterns.pm (hunk 2 above)
Comment (alvarocarvajald) [reply tests/sles4sap/patterns.pm]: Just to return the system to its previous state. But I don't think there are further tests that will require `packagekit` enabled, so I better drop it.

File: tests/sles4sap/netweaver_ascs_install.pm (hunk 3 above)
Comment (alvarocarvajald) [reply tests/sles4sap/netweaver_ascs_install.pm]: I am sure it saves time when `DESKTOP=textmode`. Not so sure in `gnome`. It also reads better semantically.

Checking the logs from the test runs I have:

With `wait_still_screen`:
- Textmode: 12 seconds
- Gnome: 19 seconds

With `check_screen`:
- Textmode: 9 seconds
- Gnome: 40 seconds

Will try the multi-tag `assert_screen` approach. Thanks.

**Comment by alvarocarvajald** [comment]
New test runs in: http://mango.suse.de/tests/249 & http://mango.suse.de/tests/250.

Elapsed time between `systemctl restart systemd-logind` and `saptune daemon status` now at:

- Textmode: 10 seconds
- Gnome: 22 seconds

Almost the same as with `wait_still_screen`.

File: tests/sles4sap/netweaver_ascs_install.pm [hunk 4]
```diff
@@ -52,8 +52,8 @@ sub run {
 
     # If running in DESKTOP=gnome, systemd-logind restart may cause the graphical console to
     # reset and appear in SUD, so need to select 'root-console' again
-    wait_still_screen;
-    select_console 'root-console';
+    assert_screen([qw(root-console displaymanager displaymanager-password-prompt generic-desktop)]);
+    select_console 'root-console' unless (match_has_tag 'root-console');
```
Comment (okurz) [inline tests/sles4sap/netweaver_ascs_install.pm]: nice, thank you :)

**Pull request merged**